### DISCUSSION TITLE
NAS organization

### DISCUSSION THREAD

**Discussion by davidk21770** [OpenSageTV/Community#166] [General]
I am / have moved from many local hard disks to 2 NASs (at 1st it was just 1 but I outgrew it <sigh> it SEEMED big!)
I need guidance on how to organize.

Since moving I am geting a lot of long spinning circles and have found that Sage seems to time-out reading the NASs main subdirs and doesn't get all of the shows and doesn't check subdirs on the one with the most shows.

I couldn't find anything in the log files that indicated an error/Error/ERROR getting files.  But when I started reorganizing to archiving some shows to import subdirs of a different folder on the NAS, it started finding things again.

What's the best way to lay things out for Sage so that it works efficiently?
Right now, I am reorganizing to have a main recording folder on each NAS and move shows to show specific subdirs in an import folder on each NAS now and as they collect.

Is this sensible?
Is there a number of files "llimit" that I should consider?  Is it by import subdir or by individual (show) subdir?

Thanks in advance for ANY and ALL suggestions or leads!

David, a long time sage user -- hoping for many more :-)

**Comment by newyankee** [comment]
Just FYI, I have one NAS dedicated to sage, and archive everything into a single directory. I cannot find any performance difference from new media on a local server disk. However, I suspect NAS cpu performance and networking issues could be a larger factor. If you have a wireless network connection, suggest only wired network configurations.

**Comment by davidk21770** [comment]
Ok.  I did the major reorganization moving most of my shows out of sage's recording directories and into the import directories and sage has sped up considerably.  I did not lose any shows or accessibility to any of the shows.  It may not be sage, it may be windows.  But I am no longer seeing long-long spinning circles that occasionally require me to close sagetv (not sage server) from a separate window.  (Not that spinning circles went away ;-) -- they are just fairly short and much less frequent.)  If I was not forced into some windows-only programs, I'd move to linux.  Maybe it's time to move sagetv to it's own box under linux...

**Comment by LoisGNS** [comment]
How do the shows stay classified as recordings, with all their metadata intact if you move them to other locations?

**Comment by davidk21770** [comment]
Sage will try to match new files to files that it has already identified.  I don't know whether it'll only match to files that it can't find.  Most of my moves just work.  Sometimes even my commercial cuts that also change the video container from .ts/.mpg to .mkv.  (I don't use .my files -- for me, they seemed to cause more problems in the long run.  When I did use them, I edited them to match the file extension change as part of the commercial cut process.  I do change the .properties to match, but I do not believe that Sage uses them -- they are for us...)

But in cases like this, where I'm reorganizing, I add the line "advanced_video_file_recovery=true" to ...SageTV/SageTV/Sage.properties (with SageTVService stopped!).
This tell sage to go to extra lengths to find moved files.  It also tells sage to take longer to age-out files that it isn't finding.
I also tell sage to check for new imported media.
After a few days, I reverse this and set "advanced_video_file_recovery=false" (after a final manual check for new imported media).

When you do move files, make sure that you move all of the associated pieces (.properties, .my, .srt, .edl or what ever commercial cut file type you're using).

There used to be a sticky in the sage discussions on how it handles these things -- it has been a LONG time since I've hunted for it.
If you don't see it -- or enough info (I'd search for advanced_video_file_recovery ;-)) -- let me know and I'll see if I can resurrect more info from my how-to notes.

**Reply by davidk21770** [reply]
oops...
The one thing that it may get wrong is to reset watched progress.  I wrote some scripts using sage's web interface to handle this but they're quick-and-dirty and probably specific to my system... especially since they use msys/bash/gawk to let me use a sane-to-me command line interface under windows.  Probably not sane to anyone else!
